docs/posts/ibisml/index.qmd

```diff
@@ -0,0 +1,928 @@
+please refer to this
+[doc](https://ibis-project.github.io/ibis-ml/reference/steps-encoding.html).
+
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to one-hot encode selected columns"
+ohe_step = ml.OneHotEncode(
+    [
+        "maritalst_893M",
+        "requesttype_4525192L",
+        "max_profession_152M",
+        "max_gender_992L",
+        "max_empl_industry_691L",
+        "max_housingtype_772L",
+        "max_incometype_1044T",
+        "max_cancelreason_3545846M",
+        "max_rejectreason_755M",
+        "education_1103M",
+        "max_status_219L",
+    ]
+)
+```
+
+### Handle date variables
+Calculate all the days difference between any date columns and the column `date_decision`:
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to calculate days difference between date columns and date_decision"
+date_cols = [col_name for col_name in df_train.columns if col_name[-1] == "D"]
+days_to_decision_expr = {
+    # difference in days
+    f"{col}_date_decision_diff": (
+        _.date_decision.epoch_seconds() - getattr(_, col).epoch_seconds()
+    )
+    / (60 * 60 * 24)
+    for col in date_cols
+}
+days_to_decision_step = ml.Mutate(days_to_decision_expr)
+```
+Extract information from the date columns:
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to extract day and week info from date columns"
+# dow and month is set to catagoery
+expand_date_step = ml.ExpandDate(ml.date(), ["week", "day"])
+```
+
+### Handle outliers
+Capping outliers using `z-score` method:
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to cap outliers for selected columns"
+step_handle_outliers = ml.HandleUnivariateOutliers(
+    ["max_amount_1115A", "max_overdueamountmax_950A"],
+    method="z-score",
+    treatment="capping",
+    deviation_factor=3,
+)
+```
+
+### Construct recipe
+We'll construct the last mile preprocessing [recipe](https://ibis-project.github.io/ibis-ml/reference/core.html#ibis_ml.Recipe)
+by chaining all transformation steps, which will be fitted to the training dataset and later applied test datasets.
+
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to construct the recipe"
+last_mile_preprocessing = ml.Recipe(
+    expand_date_step,
+    ml.Drop(ml.date()),
+    # handle string columns
+    ohe_step,
+    ml.Drop(ml.string()),
+    # handle numeric cols
+    # capping outliers
+    step_handle_outliers,
+    step_impute_median,
+    ml.ScaleMinMax(ml.numeric()),
+    # fill missing value
+    ml.FillNA(ml.numeric(), 0),
+    ml.Cast(ml.numeric(), "float32"),
+)
+print(f"Last-mile preprocessing recipe: \n{last_mile_preprocessing}")
+```
+
+## Modeling
+After completing data preprocessing with Ibis and IbisML, we proceed to the modeling
+phase. Here are two approaches:
+
+* Use IbisML as a independent data preprocessing component and hand off the data to downstream modeling
+frameworks with various output formats:
+     - Pandas Dataframe
```

Comment:
Pandas -> pandas (per their style guide, even if at the beginning of the sentence)